### PR TITLE
143 refactor AllocationsController and UsageController

### PR DESCRIPTION
**Pull request by ylyangtw**
Issue: #143 

Here are the impacted endpoints:
- /space/api/allocations
- /space/api/allocations/groups/{groupname}
- /space/api/allocations/users/{username}
- /space/api/usage
- /space/api/usage/{username}

For testing, you can use postman. Here is an example for key/value parameters in the header for testing:
**x-auth-userinfo/{"preferred_username":"test-incore-user"}
x-auth-usergroup/{"groups": ["incore_jupyter", "incore_admin", "incore_user", "incore_coe", "incore_ncsa"]}**

Please note that the following endpoints only works for admin groups, so you should expect permission denied messages when testing with non-admin groups:
- /space/api/allocations/groups/{groupname}
- /space/api/allocations/users/{username}
- /space/api/usage/{username}

### CODE REVIEW

File: server/space-service/src/main/java/edu/illinois/ncsa/incore/service/space/controllers/UsageController.java [hunk 1]
```diff
@@ -115,15 +133,10 @@ public String getUsage(@HeaderParam("x-auth-userinfo") String userInfo) {
     @Operation(summary = "Gives the usage status of the given username.",
         description = "This will only work for admin user group.")
```
Comment (longshuicy) [inline server/space-service/src/main/java/edu/illinois/ncsa/incore/service/space/controllers/UsageController.java:134]: It's not related to what you trying to do in this PR but I'm wondering if we should follow the same pattern of allocations endpoint to change this path to ` @Path("users/{username}")`

See 
```
 @GET
    @Path("users/{username}")
    @Produces(MediaType.APPLICATION_JSON)
    @Operation(summary = "Gives the allocation status of the given username.",
    ```

Comment (longshuicy) [reply server/space-service/src/main/java/edu/illinois/ncsa/incore/service/space/controllers/UsageController.java:134]: Or the other way around change allocations to just @Path("{username}")

**Review by longshuicy — APPROVED** [review]
Endpoints work. Approve

File: server/space-service/src/main/java/edu/illinois/ncsa/incore/service/space/controllers/UsageController.java (hunk 1 above)
Comment (ylyangtw) [reply server/space-service/src/main/java/edu/illinois/ncsa/incore/service/space/controllers/UsageController.java:134]: Good catch! I agree this makes the codes more consistent. I will change it to `@Path("users/{username}")` as AllocationsController needs to handle groupname as well.

Comment (ylyangtw) [reply server/space-service/src/main/java/edu/illinois/ncsa/incore/service/space/controllers/UsageController.java:134]: I think I also need to change it in the swagger document as well

**Pull request merged**